x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/SubstringTests.java

```diff
@@ -30,7 +29,7 @@
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 
-public class SubstringTests extends AbstractScalarFunctionTestCase {
+public class SubstringTests extends AbstractFunctionTestCase {
```

Comment:
It's fine to make this change now. Good, even. But I think we need to add `anyNullIsNull` and `errorsForCasesWithoutExamples` or something like that to the tests that drop `AbstractScalarFunctionTestCase`.